zimui/src/stores/main.ts

```diff
@@ -1,52 +1,94 @@
 import { defineStore } from 'pinia'
-import axios from 'axios'
+import axios, { AxiosError }from 'axios'
 import Channel from '@/types/Channel'
 import Topic from '@/types/Topic'
 
 export type RootState = {
   channelData: Channel | null
   isLoading: boolean
   errorMessage: string
+  error: AxiosError<object> | null
 }
 export const useMainStore = defineStore('main', {
   state: () =>
     ({
       channelData: null,
       isLoading: false,
       errorMessage: '',
+      error: null,
```

Comment:
Where is it used? Do we need to store it?

Reply:
the `error` object is used to store all the information regarding the error including the error message, type, code, etc. This line initializes it.

Reply:
OK, by where do you use this value, i.e. where do you retrieve what has been stored inside this store value? I see only assignments to `this.error` but nothing reading `this.error` value if I'm not mistaken, so you basically store something, hence consuming memory for doing nothing about it in the end, this is purely a waste and confusing.

Reply:
Yes, you are right, that we are only fetching and displaying the `main.errorMessage` value. 
Now, the reason we have the error object in the store is considering the ease of debugging of applications. It is one of the best practices to have the complete error log stored in an object so that whenever an error handling issue arises, the developer will just introduce the `console.log()` statement to get the description of the error. Otherwise, debugging will be hectic. So, I would propose to keep this object.

Reply:
I don't agree, please remove the `error` from the store for now.